selftests/lint.sh

```diff
@@ -8,7 +8,7 @@ fi
 PYLINT=$(which pylint-3 2>/dev/null || which pylint)
 
 MODULES="avocado contrib examples scripts docs optional_plugins selftests"
-${PYLINT} ${PYLINT_OPTIONS} --disable=W0212,W0511,W0703,W0707,R,C,E1101,E1120,E0401,I0011 --enable=R0205,R0401,W1601,W1602,W1603,W1604,W1605,W1606,W1607,W1608,W1609,W1610,W1611,W1612,W1613,W1614,W1615,W1616,W1617,W1620,W1621,W1622,W1623,W1624,W1625,W1626,W1627,W1628,W1629,W1630,W1634,W1635,W1636,W1637,W1638,W1639,W1640,W1642,W1643,W1644,W1645,W1646,W1647,W1648,W1649,W1650,W1651,W1652,W1653,W1654,W1655,W1656,W1657,W1658,W1659,W1660,W1661,W1662 ${MODULES}
+${PYLINT} ${PYLINT_OPTIONS} --disable=all --enable=R0205,R0401,W1601,W1602,W1603,W1604,W1605,W1606,W1607,W1608,W1609,W1610,W1611,W1612,W1613,W1614,W1615,W1616,W1617,W1620,W1621,W1622,W1623,W1624,W1625,W1626,W1627,W1628,W1629,W1630,W1634,W1635,W1636,W1637,W1638,W1639,W1640,W1642,W1643,W1644,W1645,W1646,W1647,W1648,W1649,W1650,W1651,W1652,W1653,W1654,W1655,W1656,W1657,W1658,W1659,W1660,W1661,W1662 ${MODULES}
```

Comment:
Thank you @tiagohonorato.

I understand that. The problem I see is that we will never be with a "consistent explicit"  group of checks, and always missing a few checks without knowing if it supposed to be enabled or disabled. For instance, I have a lot of  E0611 errors on my machine and also on GH CI env. Would be better to have assumptions like " ok, everything, except those, should be disabled by default."

So IMO we should stick with one approach (opt in) or with another (opt out).

Reply:
The problem with `--disable=all` is that when new checks are added to the default checks of pylint, and we bump its version, we'll never use them.

I see more value in having more coverage, than having a more explicit (and less confusing) way of describing the enabled checks.

So, I'd go with `--enable-all` followed by a larger `--disable=` list.  When pylint versions is bumped, we'd get the new checks, and if necessary, fix the caught issues (or as a last resort, disable them selectively).

Reply:
The suggestion from @clebergnu works for me too!

Reply:
Going with ``disable=``, as suggested by @clebergnu, sounds good for me as well.

@beraldoleal I was able to reproduce some `I1101` (apparently in `selftests/unit/test_utils_network.py`). I couldn't reproduce `E0611` yet, though. I will do some further testing.

Since we are talking about linting, another point that maybe would worth to be considered is to use our pylint configuration file as an input over `lint.sh` using the ``--rcfile=`` option. This way we would not need to edit two files when disabling/enabling lint codes.

Reply:
@tiagohonorato @clebergnu, @willianrampazzo I was working with this today, but I think that, although I would love to have `--enable=all`, we might have a problem with it:

```python
lint --enable=all # this will not enable *ALL* messages :(
```
This is still showing 71 codes as *disabled* (I did a double-check, and I don't have a pylintrc for this test). 

But:

```python
lint --disable=all
```
This is properly disabling *all* messages.

I understand the drawback of having new messages disabled by default with pylint versions bump, but IMO having all enabled will also give us some disabled messages. And we will need to add those messages anyway, like this (similarly what we are doing today):

```
enable=all, W1601, E1605, W1602, ....
```

So, unless I'm missing something, since some codes will be disabled this will keep giving us the impression that something is enabled, which is false. Besides the fact that the union on both keys is not the entire set of possibilities. 

Because of that, I think that an opt-in approach, although not perfect, is the best to follow here.

What do you guys think?